frontend/src/metabase-lib/parameters/utils/template-tags.ts

```diff
@@ -60,25 +60,21 @@ export function getTemplateTagParameters(
     .filter(
       tag =>
         tag.type != null &&
+        tag.type !== "card" &&
+        tag.type !== "snippet" &&
         ((tag["widget-type"] && tag["widget-type"] !== "none") ||
           tag.type !== "dimension"),
     )
     .map(tag => getTemplateTagParameter(tag, parametersById[tag.id]));
 }
 
-export function getTemplateTagsForParameters(card: Card) {
-  const templateTags: TemplateTag[] =
-    card &&
+export function getTemplateTags(card: Card): TemplateTag[] {
+  return card &&
     card.dataset_query &&
     card.dataset_query.type === "native" &&
```

Comment:
nit. `card` existence is checked above, dataset_query always should persist on card, but we can stay on the safe side and use optional chaining
```suggestion
  return card.dataset_query?.type === "native" &&
```

Reply:
not only this (I don't trust types I didn't write 😄 ) but also at `getParametersFromCard` there is an explicit check about card 

```
 if (!card) {
    return [];
  }
```

Reply:
Right, but `getTemplateTagParametersFromCard` is called by three functions other than `getParametersFromCard`, so that's not sufficient.

Reply:
Yeah I think I'll use `card?.dataset_query?.type === "native"` and leave it at that. I'm not confident in the change to remove the runtime check

Reply:
Updated in [0d03b3d](https://github.com/metabase/metabase/pull/37405/commits/0d03b3d156cb3369e0d3f53a7f1221e00bc1e7c8)